ReactSkia/core_modules/RSkImageLoader.cpp

```diff
@@ -0,0 +1,137 @@
+  if(imageSizeMap_.size()!=0) {
+    ImageSizemap::iterator itr,it;
+    for(itr = imageSizeMap_.begin(); itr != imageSizeMap_.end(); ++itr) {
+      CurlNetworking::sharedCurlNetworking()->abortRequest(itr->second);
+    }
+    imageSizeMap_.clear();
+  }
+}
+
+auto RSkImageLoaderModule::getConstants() -> std::map<std::string, folly::dynamic> {
+  return {};
+}
+
+auto RSkImageLoaderModule::getMethods() -> std::vector<Method> {
+  return {
+      Method(
+          "getSize",
+          [this] (dynamic args, CxxModule::Callback resolveBlock, CxxModule::Callback rejectBlock){
+            getImageSize(jsArgAsString(args,0),resolveBlock,rejectBlock);
+          }),
+      Method(
+          "getSizeWithHeaders",
+          [this] (dynamic args, CxxModule::Callback resolveBlock, CxxModule::Callback rejectBlock) {
+            getImageSizeWithHeaders(jsArgAsString(args, 0), jsArgAsString(args, 1), resolveBlock, rejectBlock);
+          }),
+      Method(
+          "prefetchImage",
+          [this] (dynamic args, CxxModule::Callback resolveBlock, CxxModule::Callback rejectBlock) {
+            prefetchImage(jsArgAsString(args, 0), resolveBlock, rejectBlock);
+          }),
+      Method(
+          "queryCache",
+          [this] (dynamic args, CxxModule::Callback resolveBlock, CxxModule::Callback rejectBlock) {
+            queryCache(jsArgAsArray(args, 0), resolveBlock, rejectBlock);
+          }),
+  };
+}
+
+std::string RSkImageLoaderModule::getName() {
+  return "ImageLoader";
+}
+
+void RSkImageLoaderModule::getImageSize(std::string uri, CxxModule::Callback resolveBlock, CxxModule::Callback rejectBlock) {
+  auto sharedCurlNetworking = CurlNetworking::sharedCurlNetworking();
+  std::shared_ptr<CurlRequest> remoteCurlRequest = std::make_shared<CurlRequest>(nullptr,uri,0,"GET");
+
+  auto completionCallback =  [remoteCurlRequest , resolveBlock , rejectBlock](void* curlresponseData,void *userdata)->bool {
+    CurlResponse *responseData =  (CurlResponse *)curlresponseData;
+    CurlRequest * curlRequest = (CurlRequest *) userdata;
+    decodedimageCacheData imageCacheData;
+    if(responseData && (responseData->responseBuffer!=nullptr) && (responseData->contentSize >0)) {
+      sk_sp<SkData> data = SkData::MakeWithCopy(responseData->responseBuffer,responseData->contentSize);
+      if (data){
+        sk_sp<SkImage> remoteImageData = SkImage::MakeFromEncoded(data);
+        RNS_LOG_DEBUG("Network response received success");
+
+        std::vector<dynamic> imageDimensions;
+        imageDimensions.push_back(folly::dynamic::array(remoteImageData->width(),remoteImageData->height()));
+        resolveBlock(imageDimensions);
+        
+        imageCacheData.imageData = remoteImageData;
+        imageCacheData.expiryTime = (SkTime::GetMSecs() + DEFAULT_MAX_CACHE_EXPIRY_TIME);//convert sec to milisecond 60 *1000
+        RSkImageCacheManager::getImageCacheManagerInstance()->imageDataInsertInCache(responseData->responseurl, imageCacheData);
```

Comment:
Here we are inserting encoded image, as decoding not needed to get metadata. This is fine.
Will this encoded image be replaced with decoded image, when it is used to display. If not, i have concern,  that image added as part of getSize, when used to display get decoded each time. Please do confirm on the behavior